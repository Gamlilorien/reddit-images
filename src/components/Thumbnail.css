.thumbnail {
    border: 2px #00CCCC solid;
  }
  .thumbnail:hover {
    border: 2px #99FFFF solid;
  }