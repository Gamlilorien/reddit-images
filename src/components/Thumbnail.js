import React from 'react';
import './Thumbnail.css';

const Thumbnail = (props) => {
    const entry = props.entry;
    return (
        <a href={entry.image} target="_blank" rel="noreferrer"><img src={entry.thumb} className="thumbnail" alt={entry.title} /></a>
    )
};

export default Thumbnail;