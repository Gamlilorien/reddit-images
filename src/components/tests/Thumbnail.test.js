import React from 'react';
import renderer from 'react-test-renderer';
import * as ReactDomClient from 'react-dom/client'
//import ReactDom from 'react-dom';
import Thumbnail from '../Thumbnail';

//sample entry format
let entry = {
    title: "Here Ya Go",
    thumb: "https://b.thumbs.redditmedia.com/i32pGUEIzNKdU6EVifHXopFsIW6jEkZ-aTXdyp4-KkY.jpg",
    image: "https://i.redd.it/ecy7638h8n391.jpg",
    link: "https://www.reddit.com/r/Funnypics/comments/v4u7s7/here_ya_go/"
}
const div = document.createElement("div");
const root = ReactDomClient.createRoot(div);

//afterEach(cleanup);

//make sure that the element gets created regarless of props
//Apparently ReactDom.render() is no longer supported so changing to createRoot method instead. See: https://github.com/reactwg/react-18/discussions/5
it("Thumbnail element renders", () => {
    root.render(<Thumbnail entry="" />)
})

//Now check if our src and alt attributes get the expected values (ie props)
//We can do this by using a snapshot. That way if the logic of the component changes it will throw an error.
it("Thumbnail wmatches test snapshot", () => {
    const tree = renderer.create(<Thumbnail entry={entry} />)
    expect(tree).toMatchSnapshot();
})