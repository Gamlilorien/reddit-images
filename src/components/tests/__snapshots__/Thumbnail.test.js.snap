// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail wmatches test snapshot 1`] = `
<a
  alt="open full size image in a new tab"
  href="https://i.redd.it/ecy7638h8n391.jpg"
  rel="noreferrer"
  target="_blank"
>
  <img
    alt="Here Ya Go"
    className="thumbnail"
    rel="noreferrer"
    src="https://b.thumbs.redditmedia.com/i32pGUEIzNKdU6EVifHXopFsIW6jEkZ-aTXdyp4-KkY.jpg"
    target="_blank"
  />
</a>
`;
